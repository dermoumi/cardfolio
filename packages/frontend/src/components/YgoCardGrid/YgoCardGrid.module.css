.ygo-card-grid {
  --max-card-width: 80px;

  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(var(--max-card-width), 1fr));
  gap: 0.2rem;
}
