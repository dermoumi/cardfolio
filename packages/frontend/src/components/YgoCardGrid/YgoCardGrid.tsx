import type { YgoCard } from "@/types";
import type { FC } from "react";

import YgoCardItem from "../YgoCardItem";
import styles from "./YgoCardGrid.module.css";

export type YgoCardGridProps = {
  cards: Array<YgoCard>;
};

const YgoCardGrid: FC<YgoCardGridProps> = ({ cards }) => {
  return (
    <div className={styles.ygoCardGrid}>
      {cards.map((card) => <YgoCardItem card={card} key={card.id} />)}
    </div>
  );
};

export default YgoCardGrid;
