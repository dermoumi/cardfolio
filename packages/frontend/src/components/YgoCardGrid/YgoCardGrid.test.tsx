import type { YgoCard } from "@/types";

import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import YgoCardGrid from ".";

describe("YgoCardGrid", () => {
  const mockCards: Array<YgoCard> = [
    {
      "id": 15,
      "updatedAt": "2025-09-03T21:54:04.817564Z",
      "name": "7 Colored Fish",
      "description": "A rare rainbow fish that has never been caught by mortal man.",
      "kind": "monster",
      "password": "23771716",
      "konamiId": 4446,
      "tcgDate": "2002-06-26",
      "ocgDate": "2000-01-27",
      "monsterKind": "normal",
      "monsterAttribute": "water",
      "monsterRace": "fish",
      "monsterAtk": 1800,
      "monsterDef": 800,
      "monsterLevel": 4,
    },
    {
      "id": 10,
      "updatedAt": "2025-09-03T21:54:02.391850Z",
      "name": "1st Movement Solo",
      "description":
        "If you control no monsters: Special Summon 1 Level 4 or lower \"Melodious\" monster from your hand or Deck. You can only activate 1 \"1st Movement Solo\" per turn. You cannot Special Summon monsters during the turn you activate this card, except \"Melodious\" monsters.",
      "kind": "spell",
      "password": "44256816",
      "konamiId": 11391,
      "tcgDate": "2014-11-06",
      "ocgDate": "2014-07-19",
      "spellKind": "normal",
    },
  ];

  it("matches snapshot", () => {
    const { asFragment } = render(<YgoCardGrid cards={mockCards} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
