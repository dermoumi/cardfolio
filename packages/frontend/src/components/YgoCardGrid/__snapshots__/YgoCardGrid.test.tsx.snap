// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YgoCardGrid > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_ygoCardGrid_898a22"
  >
    <div
      class="_ygoCardItem_b4a956"
    >
      <img
        alt="7 Colored Fish"
        src="/api/v1/ygo/cards/15/image"
      />
    </div>
    <div
      class="_ygoCardItem_b4a956"
    >
      <img
        alt="1st Movement Solo"
        src="/api/v1/ygo/cards/10/image"
      />
    </div>
  </div>
</DocumentFragment>
`;
